xbmc/ServiceManager.cpp

```diff
@@ -68,6 +68,9 @@ bool CServiceManager::InitForTesting()
   m_network.reset(new CNetwork(*m_settings));
 
   m_profileManager.reset(new CProfilesManager(*m_settings));
+  CProfile profile("special://temp");
+  m_profileManager.get()->AddProfile(profile);
+  m_profileManager.get()->CreateProfileFolders();
```

Comment:
Can this be moved to the constructor of CProfilesManager?

Reply:
`CreateProfileFolders` requires that a profile has already been added/loaded, so I don't see a possibility to call it in the constructor of `CProfilesManager`.